### PR TITLE
APPDEV-1255 Fix for buddy timeline crash issue while getting the buddy goals from stored user object.

### PR DESCRIPTION
**Pull request by Sivateja-CMI**
-- Added get user call in timeline to get updated the user with new buddies and their goals.
-- Minor refactoring.

### CODE REVIEW

File: app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java [hunk 1]
```diff
@@ -1148,19 +1141,19 @@ public YonaBuddy findYonaBuddy(Href yonaBuddy)
...
 	{
 		User user = getAppUser();
-		if (!isUserWithBuddies(user))
-		{
-			return null;
-		}
 		List<YonaBuddy> yonaBuddies = user.getBuddies();
 		for (YonaBuddy buddy : yonaBuddies)
 		{
 			if (buddy.getYonaGoals() != null)
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java]: This never returns null, so the condition can be removed.

File: app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java [hunk 2]
```diff
@@ -1148,19 +1141,19 @@ public YonaBuddy findYonaBuddy(Href yonaBuddy)
...
-				return findMatchingGoalFromYonaBuddy(goalHref, buddy);
+				YonaGoal matchingGoal = findMatchingGoalFromYonaBuddy(goalHref, buddy);
+				if (matchingGoal != null)
+				{
+					return matchingGoal;
+				}
 			}
 		}
-		return null; // Dummy return value, to allow use as data load handler
+		return null;
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java]: This is invalid. If we can't find the goal, something is badly wrong and it needs to be flagged right here. Please throw an exception like this:
```
throw new IllegalStateException("Buddy goal not found: " + goalHref.getHref());
```

File: app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java [hunk 3]
```diff
@@ -1425,7 +1410,7 @@ private void filterAndUpdateWithBuddyData(EmbeddedYonaActivity embeddedYonaActiv
 				listener.onError(new ErrorMessage(mContext.getString(R.string.no_data_found)));
 			}
 		}
-		catch (NullPointerException e)
+		catch (Exception e)
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java]: As we're not expecting any exception here, don't catch.

File: app/src/main/java/nu/yona/app/ui/BaseActivity.java [hunk 4]
```diff
@@ -34,35 +32,32 @@
...
 
 	/**
 	 * Show loading view.
 	 *
 	 * @param loading the loading
 	 * @param message the message
 	 */
-	public void showLoadingView(boolean loading, String message)
+
+	public void toggleLoadingView(boolean loading, String message)
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/BaseActivity.java]: The method name is better, but it's still wrong. Toggle suggests you're hiding if it it's visible and showing it when it's not visible.

Given that you've removed all logic (which was a good action!), just change the callers: make then call ``displayLoadingView`` or ``dismissLoadingView`` depending on their needs.

File: app/src/main/java/nu/yona/app/ui/LaunchActivity.java [hunk 5]
```diff
@@ -131,7 +131,7 @@ private void ensureValidUserEntity()
 			User user = APIManager.getInstance().getAuthenticateManager().getUser();
 			URL environmentURL = new URL(YonaApplication.getEventChangeManager().getDataState().getServerUrl());
 			URL storedUserURL = new URL(user.getLinks().getSelf().getHref());
-			if (environmentURL.getProtocol() != storedUserURL.getProtocol())
+			if (!environmentURL.getProtocol().equals(storedUserURL.getProtocol()))
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/LaunchActivity.java]: Oops! I didn't see this one. Is it wrong in the ``master`` too?

File: app/src/main/java/nu/yona/app/ui/BaseActivity.java [hunk 6]
```diff
@@ -34,35 +32,32 @@
...
+				displayLoadingView();
 			}
 			else if (progressDialog != null && !loading)
 			{
-				progressDialog.dismiss();
-				progressDialog = null;
+				dismissLoadingView();
 			}
 		}
 		catch (Exception e)
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/BaseActivity.java]: As we're not expecting an exception here, we shouldn't be catching it. Higher layers will take care of it.

File: app/src/main/java/nu/yona/app/ui/BaseActivity.java (hunk 4 above)
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/BaseActivity.java]: The parameter ``message`` isn't used, so remove it.

File: app/src/main/java/nu/yona/app/ui/LaunchActivity.java (hunk 5 above)
Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/ui/LaunchActivity.java]: Yes it is in master too.

Comment (Bert-R) [reply app/src/main/java/nu/yona/app/ui/LaunchActivity.java]: In that case, can you make a PR on master for this?

Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/ui/LaunchActivity.java]: But I doubt if there will be any harm with it. As both the references will never be equal and app will make an extra getUser call.

Comment (Bert-R) [reply app/src/main/java/nu/yona/app/ui/LaunchActivity.java]: But that'll cause extra overhead to everyone. It's better to make that 8-character PR. Pushing that to production is a very small effort.

Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/ui/LaunchActivity.java]: ok.

**Pull request merged**